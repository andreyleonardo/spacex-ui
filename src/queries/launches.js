import gql from 'graphql-tag';

export default gql`{
  launches {
    mission_name
    launch_year
  }
}
`
