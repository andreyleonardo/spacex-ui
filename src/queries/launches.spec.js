import { launchCardDetails, launchDetails } from './launches';

describe('Launches Queries', () => {
  it('returns a launchCardDetails query', () => {
    expect(launchCardDetails).toMatchSnapshot();
  });
  it('returns a launchDetails query', () => {
    expect(launchDetails).toMatchSnapshot();
  });
});
