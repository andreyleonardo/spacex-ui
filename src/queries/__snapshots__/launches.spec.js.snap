// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launches Queries returns a launchCardDetails query 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": undefined,
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "launches",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "directives": Array [],
                  "kind": "FragmentSpread",
                  "name": Object {
                    "kind": "Name",
                    "value": "launchFields",
                  },
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "launch_date_local",
                  },
                  "selectionSet": undefined,
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [],
    },
    Object {
      "directives": Array [],
      "kind": "FragmentDefinition",
      "name": Object {
        "kind": "Name",
        "value": "launchFields",
      },
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "flight_number",
            },
            "selectionSet": undefined,
          },
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "mission_name",
            },
            "selectionSet": undefined,
          },
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "launch_success",
            },
            "selectionSet": undefined,
          },
        ],
      },
      "typeCondition": Object {
        "kind": "NamedType",
        "name": Object {
          "kind": "Name",
          "value": "Launch",
        },
      },
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 159,
    "start": 0,
  },
}
`;

exports[`Launches Queries returns a launchDetails query 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "Launch",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "flight_number",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "flight_number",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "launch",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "directives": Array [],
                  "kind": "FragmentSpread",
                  "name": Object {
                    "kind": "Name",
                    "value": "launchFields",
                  },
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "launch_year",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "rocket",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "rocket_id",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "rocket_name",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "rocket_type",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": undefined,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Int",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "flight_number",
            },
          },
        },
      ],
    },
    Object {
      "directives": Array [],
      "kind": "FragmentDefinition",
      "name": Object {
        "kind": "Name",
        "value": "launchFields",
      },
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "flight_number",
            },
            "selectionSet": undefined,
          },
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "mission_name",
            },
            "selectionSet": undefined,
          },
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "launch_success",
            },
            "selectionSet": undefined,
          },
        ],
      },
      "typeCondition": Object {
        "kind": "NamedType",
        "name": Object {
          "kind": "Name",
          "value": "Launch",
        },
      },
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 313,
    "start": 0,
  },
}
`;
